src/polygonfill2d.jl

```diff
@@ -0,0 +1,98 @@
+
+"""
+    (f::BoundaryFill)(res::AbstractArray{T,2}, verts::Vector{CartesianIndex{2}}, x::Int, y::Int, fill_value::T, boundary_value::T) where {T <: Colorant}
+
+"""
+
+function (f::BoundaryFill)(
+    res::AbstractArray{T,2},
+    verts::Vector{CartesianIndex{2}},
+    x::Int,
+    y::Int,
+    fill_value::T,
+    boundary_value::T,
+    ) where {T<:Colorant}
+    if checkbounds(Bool, res, y, x)
+        if (res[y, x] != f.boundary_value && res[y, x] != f.fill_value)
+            res[y, x] = f.fill_value
+            f(res, verts, x + 1, y, fill_value, boundary_value)
+            f(res, verts, x, y + 1, fill_value, boundary_value)
+            f(res, verts, x - 1, y, fill_value, boundary_value)
+            f(res, verts, x, y - 1, fill_value, boundary_value)
+        end
```

Comment:
> You don't need to pass these to f() because it's already known to f.

What's the difference between `f.fill_value` and `fill_value`?

Reply:
They aren't different, (f.x , f.y ) and (x,y) are different as former is initial value and later pair keeps changing.